### PR TITLE
Add back the get --single functionality that has been lost during a refactoring

### PR DESCRIPTION
**Pull request by rklaehn**
Not sure if this should even be the same command. Maybe get blob or get-blob? But we have a bit of a feature explosion.

get (hash / ticket) x (collection / blob)

In any case, the single code path is now separate, which makes things simpler. One question I have is how much, if any, progress we should have when getting single or collection to stdout.

If you just get to stdout without redirecting, the stderr progress and stdout content get mixed and will look very confusing...

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -875,18 +857,129 @@ async fn get_to_dir(get: GetInteractive, out_dir: PathBuf) -> Result<()> {
...
+            .context("unable to create directory {temp_dir}")?;
+        tokio::fs::create_dir_all(&out_dir)
+            .await
+            .context("Unable to create directory {out_dir}")?;
+    }
+    let query = RangeSpecSeq::new([range]);
+    let pb = make_download_pb();
+
+    // collection info, in case we won't get a callback with is_root
+    let collection_info = Some((1, 0));
```
Comment (flub) [inline src/main.rs:900]: this is rather cryptic, the comment doesn't really help and the type is so generic I'm not sure what it describes.

File: src/main.rs [hunk 2]
```diff
@@ -875,18 +857,129 @@ async fn get_to_dir(get: GetInteractive, out_dir: PathBuf) -> Result<()> {
...
+        GetInteractive::Ticket {
+            ticket,
+            keylog,
+            derp_map,
+        } => get::run_ticket(&ticket, request, keylog, derp_map).await?,
+        GetInteractive::Hash { opts, .. } => get::run(request, opts).await?,
+    };
+    let connected = response.next().await?;
+    progress!("{} Requesting ...", style("[2/3]").bold().dim());
+    if let Some((count, missing_bytes)) = collection_info {
```
Comment (flub) [inline src/main.rs:913]: i don't see how `collection_info` would have been modified by now so isn't this essentially hardcoded?

File: src/main.rs [hunk 3]
```diff
@@ -875,18 +857,129 @@ async fn get_to_dir(get: GetInteractive, out_dir: PathBuf) -> Result<()> {
...
+        init_download_progress(&pb, count, missing_bytes);
+    }
+    let ConnectedNext::StartRoot(curr) = connected.next().await? else {
+        anyhow::bail!("Unexpected StartChild or Closing");
+    };
```
Comment (flub) [inline src/main.rs:918]: a let-else in the wild!  nice (/me still learning when they're useful)

File: src/main.rs [hunk 4]
```diff
@@ -1027,41 +1124,31 @@ async fn get_to_dir(get: GetInteractive, out_dir: PathBuf) -> Result<()> {
...
-            .with_key(
-                "eta",
-                |state: &ProgressState, w: &mut dyn std::fmt::Write| {
-                    write!(w, "{:.1}s", state.eta().as_secs_f64()).unwrap()
-                },
-            )
-            .progress_chars("#>-"),
-    );
+/// Get into a file or directory
+async fn get_to_dir(get: GetInteractive, out_dir: PathBuf) -> Result<()> {
```
Comment (flub) [inline src/main.rs:1128]: same consistency of doc comment and param name

**Review by flub — APPROVED** [review]

**Comment by flub** [comment]
wow, everything is green?!?

File: src/main.rs (hunk 2 above)
Comment (rklaehn) [reply src/main.rs:913]: yeah, this code has been split into the single/multi part. I guess there is a bunch of stuff that can go. WIll fix.

File: src/main.rs (hunk 3 above)
Comment (rklaehn) [reply src/main.rs:918]: I use them as replacement for the ugly

```rust
let x = match foo {
  Case1(x) => x,
  _ => return
}
```

stuff

**Pull request merged**